manta-pay/src/crypto/ecc/arkworks.rs

```diff
@@ -516,3 +517,83 @@ where
         )
     }
 }
+
+impl<C, CV> FixedBaseScalarMul<Compiler<C>> for GroupVar<C, CV>
+where
+    C: ProjectiveCurve,
+    CV: CurveVar<C, ConstraintField<C>>,
+{
+    type Base = C;
+
+    fn fixed_base_scalar_mul<I, T>(
+        precomputed_bases: I,
+        scalar: &Self::Scalar,
+        compiler: &mut Compiler<C>,
+    ) -> Self
+    where
+        I: IntoIterator<Item = T>,
+        T: Borrow<Self::Base>,
+    {
+        let _ = compiler;
+        let mut result = CV::zero();
+        let scalar_bits = scalar
+            .0
+            .to_bits_le()
+            .expect("Bit decomposition is not allowed to fail.");
+        for (bit, base) in scalar_bits.into_iter().zip(precomputed_bases.into_iter()) {
+            result = bit
+                .select(&(result.clone() + *base.borrow()), &result)
+                .expect("Conditional select is not allowed to fail. ");
+        }
+        Self::new(result)
```

Comment:
We eventually want this in ECLAIR. Moved to this issue:

Reply:
mark as resolved for now. We can discuss it later in issue